sources/web/datalab/polymer/components/datalab-data/datalab-data.ts

```diff
@@ -18,208 +18,170 @@
-  // private _collectionsFilterValue : string;   //TODO(jimmc): uncomment when ready to use
-
-  /** The value the user can enter to filter the tables. */
-  // private _tablesFilterValue : string;    //TODO(jimmc): uncomment when ready to use
-
-  private _collectionsList: Collection[];
-  private _tablesList: Table[];
+  private _isDetailsPaneToggledOn: boolean;
+  private _resultsList: Result[];
+  private _searchValue: string;
 
   static get is() { return 'datalab-data'; }
 
   static get properties() {
     return {
-      _collectionsFilterValue: {
-        type: String,
-        value: 'no-filter',
+      _isDetailsPaneToggledOn: {
+        type: Boolean,
+        value: true,
       },
-      _collectionsList: {
-        observer: '_renderCollectionsList',
+      _resultsList: {
+        observer: '_renderResultsList',
         type: Array,
         value: () => [],
       },
-      _tablesFilterValue: {
+      _searchValue: {
         type: String,
-        value: 'no-filter',
-      },
-      _tablesList: {
-        observer: '_renderTablesList',
-        type: Array,
-        value: () => [],
+        value: '',
       },
     };
   }
 
   ready() {
     super.ready();
 
-    this._collectionsList = [];
-    const collectionsElement = this.$.collections as ItemListElement;
-    if (collectionsElement) {
-      collectionsElement.addEventListener('itemDoubleClick',
-                                    this._collectionsDoubleClicked.bind(this));
-      collectionsElement.addEventListener('selected-indices-changed',
-                                    this._collectionsSelectionChanged.bind(this));
+    this._resultsList = [];
+    const resultsElement = this.$.results as ItemListElement;
+    if (resultsElement) {
+      resultsElement.addEventListener('itemDoubleClick',
+                                    this._resultsDoubleClicked.bind(this));
+      resultsElement.addEventListener('selected-indices-changed',
+                                    this._resultsSelectionChanged.bind(this));
     }
 
-    this._tablesList = [];
-    const tablesElement = this.$.tables as ItemListElement;
-    if (tablesElement) {
-      tablesElement.addEventListener('itemDoubleClick',
-                                    this._tablesDoubleClicked.bind(this));
-      tablesElement.addEventListener('selected-indices-changed',
-                                    this._tablesSelectionChanged.bind(this));
-    }
+    (this.$.results as ItemListElement).columns = ['Name', 'Type'];
+    this.$.searchKeys.target = this.$.searchBox;
   }
 
-  _generateFakeCollectionsListForTesting() {
-    const collectionsList = [];
-    const count = Math.floor(Math.random() * 20);
-    for (let i = 0; i < count; i++) {
-      collectionsList.push(this._generateFakeCollectionForTesting());
-    }
-    return collectionsList;
+  /** Sends the user's query to the search API, renders results as they get returned. */
+  _search() {
+    this._resultsList = [];
```

Comment:
This could make an unnecessary UI refresh if the list is unchanged. Can we move it to the `_handleQueryResults` where we don't do anything if the results are the same? I assume this will happen a lot as the user is writing a query to narrow down results.
Also, I'm not sure how polished you want this code to be now, so feel free to ignore comments like this, but it'd be good to at least add a TODO to revisit later.